src/io/record-play-movie.cpp

```diff
@@ -167,56 +188,87 @@ static bool string_is_repeat(char *str, int len)
     return true;
 }
 
-static errr send_text_to_chuukei_server(TERM_LEN x, TERM_LEN y, int len, TERM_COLOR col, concptr str)
+#ifdef JP
+/* マルチバイト文字を分割せずに str を分割する長さを len 以下で返す */
+static int find_split(concptr str, int len)
 {
-    char buf[1024 + 32];
-    char buf2[1024];
-
-    strncpy(buf2, str, len);
-    buf2[len] = '\0';
+    /*
+     * SJIS が定義されている場合でも、str は常に EUC-JP としてエンコードされます。
+     * 他の場所で行われているような 3 バイトのエンコーディングは想定していません。
+     */
+    int last_split = 0, i = 0;
+    while (i < len) {
+        if (((unsigned char)str[i] >= 0xa1 && (unsigned char)str[i] <= 0xfe) || (unsigned char)str[i] == 0x8e) {
```

Comment:
This block is similar to is\_kanji() function macro.
It seems to be better to define the macro or inline global function in h-config.h.

Reply:
The latest forced push defines iseuckanji() in h-config.h if JP is set on any platform.  It then defines iskanji() in terms of iseuckanji() on the platforms which don't use Shift-JIS.  record-play-movie.cpp:206 uses iseuckanji().